design/FY2021/NFP-Spaces.md

```diff
@@ -0,0 +1,234 @@
+# Adding "Spaces" to EnergyPlus #
+
+**Michael J. Witte, GARD Analytics, Inc.**  
+**Jason W. DeGraw, ORNL**  
+**With input from many more . . .**  
+
+ -  Original, November 22, 2019
+ -  Revised, April 1, 2020
+    * Keep the current definition of Zone
+    * Add new objects for Space, SpaceType, CompoundSpaceType, and Enclosure
+ -  Revised, November 25, 2020
+    * Condensed and focused on final proposal somewhat agreed to back in April 2020
+    * Revised to reflect simplified Construction:AirBoundary object (radiant and solar enclosures are now the same)
+
+## Justification for New Feature ##
+
+Thermal zones (HVAC zones) are often composed of a variety of spaces grouped together to be served by
+a single thermostat. For example, an office building core zone may include private offices, conference rooms, restrooms
+and corridors. Each of these spaces (rooms) has different internal gains and may or may not be fully enclosed by solid surfaces.
+Currently, the smallest building element is a zone. This requires the user (or interface) to blend these spaces together for surfaces,
+internal gains, and other specifications. The option to specify each space explicitly would greatly simplify input in data managemenent,
+especially for space-based interfaces (such as OpenStudio) and for codes and standards modeling.
+
+There are also computing performance advantages to dividing zones into smaller enclosures for radiant exchange.
+
+## Overview ##
+
+### Current EnergyPlus model ###
+EnergyPlus input currently has the following (very flat) heierarchy:
+
+    Zone
+        Surfaces (references a zone name)
+        People, Lights, etc. (references a zone name or zone list name)
+        Thermostat (one per zone)
+        HVAC equipment (attaches to a zone)
+
+Zones represent an air mass (air node) that exchanges heat with surfaces, internal loads, HVAC equipment, etc. 
+A Zone is essentially the "atomic unit" of the building. By default a zone has a uniform air temperature, but there are room air models 
+that allow modeling of stratification and other non-uniform temperature effects. Thermostats and HVAC equipment are assigned per Zone.
+
+The internal data model adds another layer:
+
+    Enclosure
+        Zone
+            Surfaces
+            People, Lights, etc.
+            Thermostat
+            HVAC equipment
+        Zone
+            Surfaces
+            People, Lights, etc.
+            Thermostat
+            HVAC equipment
+
+Enclosures are used for radiant and solar/daylighting exchange. Enclosures are primarily concerned with Surfaces, but they
```

Comment:
To me Enclosure would imply a fully-enclosed volume, is that right?  Are Zone's not fully-enclosed?  What is the relationship between Zone's and Enclosure's?  Does a Zone have an Enclosure?  Can a Zone have multiple Enclosures?  Can an Enclosure be shared by two Zones?

Reply:
Are Spaces fully-enclosed too?